ddl/db_test.go

```diff
@@ -3663,14 +3663,14 @@ func (s *testDBSuite4) TestIfNotExists(c *C) {
 	tk.MustGetErrCode(sql, errno.ErrDupKeyName)
 	s.mustExec(tk, c, "alter table t1 add index if not exists idx_b (b)")
 	c.Assert(tk.Se.GetSessionVars().StmtCtx.WarningCount(), Equals, uint16(1))
-	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Note|1061|index already exist idx_b"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Note|1061|index already exist idx_b; a background job is trying to add the same index, please check by `ADMIN SHOW DDL JOBS`"))
```

Comment:
How can you ensure that the job before is still running?

Reply:
You can neither ensure that the job is finished or not, so either message would make the test case unstable. The only sure thing is that the message starts with "Note|1061|index already exist idx_b;".

Reply:
Besides, you can do a manual test and paste it in the PR description.

Reply:
> You can neither ensure that the job is finished or not, so either message would make the test case unstable. The only sure thing is that the message starts with "Note|1061|index already exist idx_b;".

Got it, thx a lot. I'm not sure whether it can check the prefix of the error message; or, on the other hand, is it necessary to add test cases?

Reply:
It's not necessary to add more unit test cases. But you must modify this test case, since you have modified the error message.
Besides, it's better to do a manual test when it's hard to add a unit test.

Reply:
For the test here, it can be sure that after line 3664(`s.mustExec(tk, c, "alter table t1 add index if not exists idx_b (b)")`) is executed, the job has been done. It should be stable.

For the test case, you can either:
- Do a manual test, as suggested by @djshow832 since the change is trivial enough.
- Create a new test case that:
  1. Create a table with an index, say `create table t(a int, key idx(a));`
  2. Get the meta info of index `idx` and modify it to a state rather than `StatePublic` internally.
  3. Create an index with name `idx` and expect the explicit error message.

Both ways are fine to me, it's up to you to choose any of them.

Reply:
You can get the warnings by `StmtCtx.GetWarnings` and then check the error code.